tilelang/jit/adapter/tvm_ffi.py

```
@@ -209,7 +209,10 @@ def func(*inputs: torch.Tensor | Any):
 
             # Resolve the device used for outputs. Prefer the first tensor input's device
             # if available, otherwise use PyTorch's current device.
-            out_device: torch.device | None = None
+            out_device: torch.device | None = next(
+                (input.device for input in inputs if isinstance(input, torch.Tensor)),
```

Ruff (0.15.17)

[error] Variable `input` is shadowing a Python builtin

(A001)